firebase-storage/ktx/src/main/kotlin/com/google/firebase/storage/ktx/Storage.kt

```diff
@@ -141,80 +142,22 @@ operator fun ListResult.component2(): List<StorageReference> = prefixes
 operator fun ListResult.component3(): String? = pageToken
 
 /**
- * Starts listening to this task's stream download progress and emits its values via a [Flow].
+ * Starts listening to this task's progress and emits its values via a [Flow].
  *
  * - When the returned flow starts being collected, it attaches the following listeners:
- * [OnPausedListener], [OnProgressListener].
+ * [OnProgressListener], [OnPausedListener].
```

Comment:
@vkryachko sounds like we'd need to add an `OnCompleteListener`:
```kotlin
        val completionListener = OnCompleteListener<T> { task ->
            if (task.isSuccessful) {
                close()
            } else {
                val exception = task.exception
                cancel("Error getting TaskState", exception)
            }
        }
```

Do you think we should also add an early close at the beginning of the flow? In case the Flow starts collecting after the task has already completed.
```kotlin
    get() = callbackFlow {
        if (isComplete) {
            if (isSuccessful) {
                close()
            } else {
                val exception = exception
                cancel("Error getting TaskState", exception)
            }
        }
        // ... (add listeners) ...
    }
```

Reply:
> Do you think we should also add an early close at the beginning of the flow?

I don't think it's necessary, adding an `OnCompleteListener` will fire immediately if the task is completed.

Reply:
@vkryachko Sounds good to me. Added the `OnCompleteListener`.